test/components/interpolatedColorLegendTests.ts

```diff
@@ -3,113 +3,114 @@
-    let swatchContainerBCR = (<Element> swatchContainer.node()).getBoundingClientRect();
-
-    let lowerLabelBCR = (<Element> labels[0][0]).getBoundingClientRect();
-    let upperLabelBCR = (<Element> labels[0][1]).getBoundingClientRect();
-    assert.operator(swatchContainerBCR.right, "<=", lowerLabelBCR.left, "first label to right of swatches");
-    assert.operator(swatchContainerBCR.right, "<=", upperLabelBCR.left, "second label to right of swatches");
-    assert.operator(upperLabelBCR.bottom, "<=", lowerLabelBCR.top, "lower label is drawn below upper label");
-
-    svg.remove();
-  });
-
-  it("renders correctly when oriented left", () => {
-    let legend = new Plottable.Components.InterpolatedColorLegend(colorScale);
-    legend.orientation("left");
-    legend.renderTo(svg);
-
-    assertBasicRendering(legend);
-
-    let legendElement: d3.Selection<void> = (<any> legend)._element;
-    let labels = legendElement.selectAll("text");
-    let swatchContainer = legendElement.select(".swatch-container");
-    let swatchContainerBCR = (<Element> swatchContainer.node()).getBoundingClientRect();
-
-    let lowerLabelBCR = (<Element> labels[0][0]).getBoundingClientRect();
-    let upperLabelBCR = (<Element> labels[0][1]).getBoundingClientRect();
-    assert.operator(lowerLabelBCR.left, "<=", swatchContainerBCR.left, "first label to left of swatches");
-    assert.operator(upperLabelBCR.left, "<=", swatchContainerBCR.left, "second label to left of swatches");
-    assert.operator(upperLabelBCR.bottom, "<=", lowerLabelBCR.top, "lower label is drawn below upper label");
+  describe("rendering", () => {
+    HORIZONTAL_ORIENTATIONS.forEach((orientation: string) => {
+      it(`renders swatches and labels in correct order when oriented ${orientation}`, () => {
+        const legend = new Plottable.Components.InterpolatedColorLegend(colorScale).orientation(orientation);
+        legend.renderTo(svg);
+
+        assertBasicRendering(legend);
+
+        const legendElement: d3.Selection<void> = (<any> legend)._element;
+        const labels = legendElement.selectAll("text");
+        const swatchContainer = legendElement.select(".swatch-container");
+        const swatchContainerBCR = (<Element> swatchContainer.node()).getBoundingClientRect();
+
+        const lowerLabelBCR = (<Element> labels[0][0]).getBoundingClientRect();
+        const upperLabelBCR = (<Element> labels[0][1]).getBoundingClientRect();
+        assert.closeTo(lowerLabelBCR.left, swatchContainerBCR.left, window.Pixel_CloseTo_Requirement, "left align first label ");
+        assert.closeTo(upperLabelBCR.right, swatchContainerBCR.right, window.Pixel_CloseTo_Requirement, "right align second label");
+        if (orientation === "top") {
+          assert.operator(lowerLabelBCR.top, "<=", swatchContainerBCR.top, `first label is to the top of swatches`);
+          assert.operator(upperLabelBCR.top, "<=", swatchContainerBCR.top, `second label is to the top of swatches`);
```

Comment:
Shouldn't we check that the `bottom` of the labels is `<=` the `top` of the swatch container?